src/serving.rs

```diff
@@ -245,7 +244,10 @@ pub fn serve<E: Entity>(e: E, req: &Request) -> Response<E::Body> {
         }
     };
     if include_entity_headers {
-        e.add_headers(res.headers_mut());
+        let mut headers = http::header::HeaderMap::new();
+        e.add_headers(&mut headers);
+        let hyper_headers: hyper::header::Headers = headers.into();
```

Comment:
Did you consider changing `serve` to return a `http::Response` rather than a `hyper::server::Response` instead of converting here? I think I'd prefer to use the same interface throughout, unless there's some reason that won't work?

Reply:
Unless I missed something, switching to the `http` interface was harder because it lacks typed headers and only works with strings. So things like `hyper::header::LastModified` would no longer work.

Reply:
Ahh, fair enough. I missed your last comment on the issue. I think it shouldn't be too bad to switch over to the stringly typed stuff, but there's enough to it (e.g. changing the `Range` parsing) that it should be a separate commit.